src/exts/discord_integration.js

```diff
@@ -0,0 +1,94 @@
+const { bridge } = require('../spring_api');
+const { log } = require('../spring_log');
+const { config } = require('../launcher_config');
+const DiscordRPC = require("discord-rpc");
+
+if (config) {
+    if (config.discord_rich_presence.application_id == null) {
+        log.warn("config.discord_rich_presence.application_id not defined");
+        return
+    }
+}
+
+// Application ID from https://discord.com/developers/applications/<applicationId>/
+const applicationId = "1185990483143569448";
+const rpc = new DiscordRPC.Client({ transport: 'ipc' });
+
+DiscordRPC.register(applicationId);
+
+bridge.on('DiscordSetActivity', async command => {
+    // command
+    // {
+    //     state - string
+    //     details - string
+    //     startTimestamp - unix timestamp
+    //     playerCount - int > 0
+    //     maxPlayerCount - int > 0
+    //     partyId - string
+    //     largeImageText - string
+    //     smallImageKey - string
+    //     smallImageText - string
+    // }
+
+    if (!rpc) {
+        log.warn("DiscordSetActivity - No Discord RPC");
+        return
+    };
+
+    // Playing, spectating, in menu, in lobby, watching replay...
+    const state = command.state;
+
+    // Map name
+    const details = command.details;
+
+    // Epoch seconds for game start - including will show time as "elapsed"
+    const startTimestamp = command.startTimestamp;
+
+    // Name of the uploaded image for the large profile artwork
+    // Minimap Image URL
+    const largeImageKey = command.details ?
+        String(config.discord_rich_presence.minimap_link).replace("<map>", encodeURIComponent(command.details)) :
```

Comment:
You don't need to and shouldn't call the `String`, just do .replace on the `minimap_link`.

Reply:
I am using String in case `config.discord_rich_presence.minimap_link` is not defined in conig and is null.

Reply:
If it's null, why set it as 'null' string?